flink/interpreter/pom.xml

```diff
@@ -95,6 +96,13 @@
       <version>${project.version}</version>
     </dependency>
 
+    <dependency>
+      <groupId>io.fabric8</groupId>
+      <artifactId>kubernetes-client</artifactId>
+      <version>${kubernetes.client.version}</version>
+      <scope>compile</scope>
```

Comment:
Where have you used this dependency?

Reply:
Right，I didn't use it yet, Just plan to add ingress support in next PR. Let me remove it in this PR.

Reply:
I also have the requirement to add additional ingress k8s resources during the runtime of the zeppelin interpreter. See [ZEPPELIN-5145](https://issues.apache.org/jira/browse/ZEPPELIN-5145).

In my opinion, the interpreter pod should **not** have the right to create ingress resources, but the Zeppelin server service user should have the right.

So my plan is to add an additional interface to the Zeppelin interpreter project. Something like `ExternalURLSupported`.
This interface should be implemented by the `InterpreterProcess` e.g. `K8sRemoteInterpreterProcess.java` on the ZeppelinServer side and any specific interpreter can query the function via `instanceof`.